### PR TITLE
added "declare (strict_types = 1);" and classes are set as final

### PR DESCRIPTION
**Pull request by thenotsoft**
added "declare (strict_types = 1);", classes are set as final, and bugs found are fixed

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  |

### CODE REVIEW

File: src/Data/DataResponse.php [hunk 1]
```diff
@@ -12,7 +14,7 @@
  * For example, `['name' => 'Dmitriy']` to be formatted to JSON using {@see \Yiisoft\Yii\Web\Data\Formatter\JsonDataResponseFormatter}
  * when {@see DataResponse::getBody()} is called.
  */
-class DataResponse implements ResponseInterface
+final class DataResponse implements ResponseInterface
```
Comment (roxblnfk) [inline src/Data/DataResponse.php:17]: I think that here @yiiliveext left the opportunity to extend the decorator

Comment (thenotsoft) [reply src/Data/DataResponse.php:17]: > I think that here @yiiliveext left the opportunity to extend this decorator

We will remove "final" when necessary.

File: src/Middleware/Csrf.php [hunk 2]
```diff
@@ -91,6 +93,6 @@ private function getTokenFromRequest(ServerRequestInterface $request): ?string
             $token = \reset($headers);
         }
 
-        return TokenMask::remove($token);
+        return is_string($token) ? TokenMask::remove($token) : null;
```
Comment (samdark) [inline src/Middleware/Csrf.php:96]: In what case it won't be a string?

File: src/Middleware/HttpCache.php [hunk 3]
```diff
@@ -96,16 +98,12 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
         }
 
         $cacheIsValid = $this->validateCache($request, $lastModified, $etag);
+        $response = $handler->handle($request);
+
         if ($cacheIsValid) {
-            return $this->responseFactory
```
Comment (samdark) [inline src/Middleware/HttpCache.php:100]: @rustamwin would you please verify that the change is correct?

File: src/Middleware/Csrf.php (hunk 2 above)
Comment (thenotsoft) [reply src/Middleware/Csrf.php:96]: `reset ` return boolean if the array is empty

File: src/Middleware/HttpCache.php (hunk 3 above)
Comment (rustamwin) [reply src/Middleware/HttpCache.php:100]: That is correct, yes.

**Comment by samdark** [comment]
👍

**Pull request merged**